app/controllers/submit_form_controller.rb

```diff
@@ -98,4 +103,42 @@ def build_attachments_index(submission)
     ActiveStorage::Attachment.where(record: submission.submitters, name: :attachments)
                              .preload(:blob).index_by(&:uuid)
   end
+
+  def fetch_ats_prefill_values_if_available
+    # ATS passes values directly as Base64-encoded JSON parameters
+    return {} if params[:ats_values].blank?
+
+    # Security: Limit input size to prevent DoS attacks (64KB limit)
+    if params[:ats_values].bytesize > 65_536
+      Rails.logger.warn "ATS prefill values parameter exceeds size limit: #{params[:ats_values].bytesize} bytes"
+      return {}
+    end
+
+    begin
+      decoded_json = Base64.urlsafe_decode64(params[:ats_values])
+
+      # Security: Limit decoded JSON size as well
+      if decoded_json.bytesize > 32_768
+        Rails.logger.warn "ATS prefill decoded JSON exceeds size limit: #{decoded_json.bytesize} bytes"
+        return {}
+      end
+
+      ats_values = JSON.parse(decoded_json)
+
+      # Validate that we got a hash
+      if ats_values.is_a?(Hash)
+        # Audit logging: Log ATS prefill usage for security monitoring
+        Rails.logger.info "ATS prefill values processed for submitter: #{@submitter&.slug || 'unknown'}, " \
+                          "field_count: #{ats_values.keys.length}, " \
+                          "account: #{@submitter&.account&.name || 'unknown'}"
+        ats_values
+      else
+        Rails.logger.warn "ATS prefill values not a hash: #{ats_values.class}"
+        {}
+      end
+    rescue StandardError => e
+      Rails.logger.warn "Failed to parse ATS prefill values: #{e.message}"
+      {}
+    end
+  end
```

Comment:
Do we need all of these logs? I think the comments are enough for most of these.